internal/loader/configuration_client_manager.go

```diff
@@ -37,6 +37,8 @@ import (
 
 type ConfigurationClientManager struct {
 	ReplicaDiscoveryEnabled   bool
+	LoadBalancingEnabled      bool
+	IsFailoverRequest         bool
```

Comment:
I don't think it's required to be a property in `ConfigurationClientManager`, we don't need to remember the state of `IsFailoverRequest`